FWIO/RNTuple/plugins/RNTupleOutputModule.cc

```diff
@@ -0,0 +1,262 @@
+    bool useBufferedWrite_;
+    bool useDirectIO_;
+    bool dropMetaData_;
+    bool useStreamer_;
+  };
+
+  RNTupleOutputModule::RNTupleOutputModule(ParameterSet const& pset)
+      : one::OutputModuleBase(pset),
+        one::OutputModule<>(pset),
+        fileName_(pset.getUntrackedParameter<std::string>("fileName")),
+        overrideStreamer_(
+            fromConfig(pset.getUntrackedParameter<std::vector<edm::ParameterSet>>("overrideDataProductStreamer"))),
+        noSplitSubFields_(pset.getUntrackedParameter<std::vector<std::string>>("noSplitSubFields")),
+        compressionAlgo_(convertTo(pset.getUntrackedParameter<std::string>("compressionAlgorithm"))),
+        compressionLevel_(pset.getUntrackedParameter<unsigned int>("compressionLevel")),
+        approxZippedClusterSize_(pset.getUntrackedParameter<unsigned long long>("approxZippedClusterSize")),
+        maxUnzippedClusterSize_(pset.getUntrackedParameter<unsigned long long>("maxUnzippedClusterSize")),
+        initialUnzippedPageSize_(pset.getUntrackedParameter<unsigned long long>("initialUnzippedPageSize")),
+        maxUnzippedPageSize_(pset.getUntrackedParameter<unsigned long long>("maxUnzippedPageSize")),
+        pageBufferBudget_(pset.getUntrackedParameter<unsigned long long>("pageBufferBudget")),
+        useBufferedWrite_(pset.getUntrackedParameter<bool>("useBufferedWrite")),
+        useDirectIO_(pset.getUntrackedParameter<bool>("useDirectIO")),
+        dropMetaData_(pset.getUntrackedParameter<bool>("dropPerEventDataProductProvenance")),
+        useStreamer_(pset.getUntrackedParameter<bool>("useStreamer")) {}
+
+  void RNTupleOutputModule::openFile(FileBlock const& fb) {
+    RNTupleOutputFile::Config conf;
+    conf.wantAllEvents = wantAllEvents();
+    conf.selectorConfig = selectorConfig();
+    conf.compressionAlgo = compressionAlgo_;
+    conf.compressionLevel = compressionLevel_;
+    conf.approxZippedClusterSize = approxZippedClusterSize_;
+    conf.maxUnzippedClusterSize = maxUnzippedClusterSize_;
+    conf.initialUnzippedPageSize = initialUnzippedPageSize_;
+    conf.maxUnzippedPageSize = maxUnzippedPageSize_;
+    conf.pageBufferBudget = pageBufferBudget_;
+    conf.useBufferedWrite = useBufferedWrite_;
+    conf.useDirectIO = useDirectIO_;
+    conf.dropMetaData = dropMetaData_;
+    conf.doNotSplitSubFields = noSplitSubFields_;
+    if (useStreamer_ and overrideStreamer_.empty()) {
+      auto const& prods = keptProducts()[InEvent];
+      conf.streamerProduct = std::vector<bool>(prods.size(), true);
+    } else if (not overrideStreamer_.empty()) {
+      auto const& prods = keptProducts()[InEvent];
+      conf.streamerProduct = std::vector<bool>(prods.size(), useStreamer_);
+      unsigned int index = 0;
+      for (auto const& prod : prods) {
+        auto choice = useStreamer(prod.first->branchName(), overrideStreamer_);
+        if (choice) {
+          std::cout << "found match " << prod.first->branchName() << std::endl;
```

Comment:
If this printout still useful? If yes, change to use MessageLogger.

Reply:
removed